src/main/scala/chisel3/util/LTL.scala

```diff
@@ -0,0 +1,392 @@
+    * in SVA.
+    */
+  def implication(seq: Sequence, prop: Property): Property = {
+    val ltl_implication = Instantiate(new LTLImplicationIntrinsic)
+    ltl_implication.lhs := seq.inner
+    ltl_implication.rhs := prop.inner
+    OpaqueProperty(ltl_implication.out)
+  }
+
+  /** Non-overlapping variant of `Property.implication`. Equivalent to
+    * `seq ##1 true |-> prop` and `seq |=> prop` in SVA.
+    */
+  def implicationNonOverlapping(seq: Sequence, prop: Property): Property = {
+    import Sequence.BoolSequence
+    Property.implication(seq.concat(true.B.delay(1)), prop)
+  }
+
+  /** Indicate that a property will eventually hold at a future point in time.
+    * This is a *strong* eventually, so the property has to hold within a finite
+    * number of cycles. The property does not trivially hold by waiting an
+    * infinite number of cycles.
+    *
+    * Equivalent to `s_eventually prop` in SVA.
+    */
+  def eventually(prop: Property): Property = {
+    val ltl_eventually = Instantiate(new LTLEventuallyIntrinsic)
+    ltl_eventually.in := prop.inner
+    OpaqueProperty(ltl_eventually.out)
+  }
+
+  /** Form the conjunction of two properties. Equivalent to
+    * `arg0 and arg1 and ... and argN` in SVA.
+    */
+  def and(arg0: Property, argN: Property*): Property = {
+    var lhs = arg0
+    for (rhs <- argN) {
+      val ltl_and = Instantiate(new LTLAndIntrinsic)
+      ltl_and.lhs := lhs.inner
+      ltl_and.rhs := rhs.inner
+      lhs = OpaqueProperty(ltl_and.out)
+    }
+    lhs
+  }
+
+  /** Form the disjunction of two properties. Equivalent to
+    * `arg0 or arg1 or ... or argN` in SVA.
+    */
+  def or(arg0: Property, argN: Property*): Property = {
+    var lhs = arg0
+    for (rhs <- argN) {
+      val ltl_or = Instantiate(new LTLOrIntrinsic)
+      ltl_or.lhs := lhs.inner
+      ltl_or.rhs := rhs.inner
+      lhs = OpaqueProperty(ltl_or.out)
+    }
+    lhs
+  }
+
+  /** Specify a `clock` relative to which all cycle delays within `prop` are
+    * specified. Equivalent to `@(posedge clock) prop` in SVA.
+    */
+  def clock(prop: Property, clock: Clock): Property = {
+    val ltl_clock = Instantiate(new LTLClockIntrinsic)
+    ltl_clock.in := prop.inner
+    ltl_clock.clock := clock
+    OpaqueProperty(ltl_clock.out)
+  }
+
+  /** Disable the checking of a property if a condition is true. If the
+    * condition is true at any time during the evaluation of the property, the
+    * evaluation is aborted. Equivalent to `disable iff (cond) prop` in SVA.
+    */
+  def disable(prop: Property, cond: Bool): Property = {
+    val ltl_disable = Instantiate(new LTLDisableIntrinsic)
+    ltl_disable.in := prop.inner
+    ltl_disable.condition := cond
+    OpaqueProperty(ltl_disable.out)
+  }
+}
+
+/** The base class for the `AssertProperty`, `AssumeProperty`, and
+  * `CoverProperty` verification constructs.
+  */
+abstract class AssertPropertyLike {
```

Comment:
```suggestion
sealed abstract class AssertPropertyLike {
```
We don't want users extending this right?

Reply:
Yeah, great point!